File: Examples/test-suite/perl5/overload_null_runme.pl
use strict;
use warnings;
use Test::More tests => 24;
BEGIN { use_ok('overload_null') }
require_ok('overload_null');

my $o = new overload_null::Overload();
my $x = new overload_null::X();

is(1, $o->byval1($x));
is(2, $o->byval1(undef));

is(3, $o->byval2(undef));
is(4, $o->byval2($x));

is(5, $o->byref1($x));
is(6, $o->byref1(undef));

is(7, $o->byref2(undef));
is(8, $o->byref2($x));

is(9, $o->byconstref1($x));
is(10, $o->byconstref1(undef));

is(11, $o->byconstref2(undef));
is(12, $o->byconstref2($x));

# const pointer references
is(13, $o->byval1cpr($x));
is(14, $o->byval1cpr(undef));

is(15, $o->byval2cpr(undef));
is(16, $o->byval2cpr($x));

# forward class declaration
is(17, $o->byval1forwardptr($x));
is(18, $o->byval1forwardptr(undef));

is(19, $o->byval2forwardptr(undef));
is(20, $o->byval2forwardptr($x));

is(21, $o->byval1forwardref($x));

is(22, $o->byval2forwardref($x));

File: Examples/test-suite/octave/overload_null_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

overload_null

function check(a, b)
  if (a != b)
    error("%i does not equal %i", a, b);
  endif
end

o = Overload();
x = X();
null = []; # NULL pointer

check(1, o.byval1(x));
check(2, o.byval1(null));

check(3, o.byval2(null));
check(4, o.byval2(x));

check(5, o.byref1(x));
check(6, o.byref1(null));

check(7, o.byref2(null));
check(8, o.byref2(x));

check(9, o.byconstref1(x));
check(10, o.byconstref1(null));

check(11, o.byconstref2(null));
check(12, o.byconstref2(x));

# const pointer references
check(13, o.byval1cpr(x));
check(14, o.byval1cpr(null));

check(15, o.byval2cpr(null));
check(16, o.byval2cpr(x));

# forward class declaration
check(17, o.byval1forwardptr(x));
check(18, o.byval1forwardptr(null));

check(19, o.byval2forwardptr(null));
check(20, o.byval2forwardptr(x));

check(21, o.byval1forwardref(x));

check(22, o.byval2forwardref(x));
